File: app/src/main/java/deltazero/amarok/AmarokActivity.java
package deltazero.amarok;

import android.app.ActivityManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.WindowManager;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import deltazero.amarok.core.SettingsSnapshot;
import deltazero.amarok.ui.CalendarActivity;
import deltazero.amarok.ui.SecurityAuthActivity;
import deltazero.amarok.utils.SecurityUtil;
import java.util.List;

public class AmarokActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    // Enable edge-to-edge display
    EdgeToEdge.enable(this);
  }

  @Override
  protected void onStart() {
    SettingsSnapshot settings =
        ((AmarokApplication) getApplication()).getSettingsRepo().getSettings().getValue();
    if (settings.getBlockScreenshots())
      getWindow()
          .setFlags(WindowManager.LayoutParams.FLAG_SECURE, WindowManager.LayoutParams.FLAG_SECURE);

    if (settings.getHideFromRecents()) {
      ActivityManager am = (ActivityManager) getSystemService(Context.ACTIVITY_SERVICE);
      if (am != null) {
        List<ActivityManager.AppTask> tasks = am.getAppTasks();
        if (!tasks.isEmpty()) {
          tasks.get(0).setExcludeFromRecents(true);
        }
      }
    }

    super.onStart();
  }

  @Override
  protected void onResume() {
    AmarokApplication app = (AmarokApplication) getApplication();
    if (SecurityUtil.isDisguiseNeeded(app)) startActivity(new Intent(this, CalendarActivity.class));
    else if (SecurityUtil.isUnlockRequired(app))
      startActivity(new Intent(this, SecurityAuthActivity.class));
    super.onResume();
  }
}

File: app/src/main/java/deltazero/amarok/AmarokActivity.kt
package deltazero.amarok

import android.app.ActivityManager
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.WindowManager
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import dagger.hilt.android.AndroidEntryPoint
import deltazero.amarok.core.SettingsRepository
import deltazero.amarok.ui.CalendarActivity
import deltazero.amarok.ui.SecurityAuthActivity
import deltazero.amarok.utils.SecurityUtil
import javax.inject.Inject

@AndroidEntryPoint
open class AmarokActivity : AppCompatActivity() {
  @Inject lateinit var settingsRepo: SettingsRepository

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    // Enable edge-to-edge display
    enableEdgeToEdge()
  }

  override fun onStart() {
    val settings = settingsRepo.settings.value
    if (settings.blockScreenshots) {
      window.setFlags(
        WindowManager.LayoutParams.FLAG_SECURE,
        WindowManager.LayoutParams.FLAG_SECURE,
      )
    }

    if (settings.hideFromRecents) {
      val am = getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager?
      if (am != null) {
        val tasks = am.appTasks
        if (tasks.isNotEmpty()) {
          tasks[0].setExcludeFromRecents(true)
        }
      }
    }

    super.onStart()
  }

  override fun onResume() {
    if (SecurityUtil.isDisguiseNeeded(settingsRepo))
      startActivity(Intent(this, CalendarActivity::class.java))
    else if (SecurityUtil.isUnlockRequired(settingsRepo)) {
      startActivity(Intent(this, SecurityAuthActivity::class.java))
    }
    super.onResume()
  }
}
